.github/workflows/tests-unit-php.yml

```diff
@@ -131,7 +131,7 @@ jobs:
           composer require --dev --ignore-platform-reqs --no-interaction --no-scripts --update-with-all-dependencies phpunit/phpunit:^7.5
           git apply patches/composer.patch
           composer dump-autoload --no-interaction
-        if: ${{ matrix.php == '8.0' }}
+        if: ${{ matrix.php == '8.0' || matrix.php == '7.4' }}
```

Comment:
This is a nice workaround for PHPUnit issues. What do you think @swissspidy ?

Reply:
This could be simplified to `if: ${{ matrix.php != '7.0' }}`

I mentioned [on Slack](https://xwp.slack.com/archives/C013J2EGH8R/p1625870237152700?thread_ts=1625839119.147100&cid=C013J2EGH8R) that this workaround might make it more difficult to run the unit tests locally as one would need to patch PHPUnit first.

That means I'd have to run all this locally first so I could run the tests:

```bash
composer require --dev --ignore-platform-reqs --no-interaction --no-scripts --update-with-all-dependencies phpunit/phpunit:^7.5
git apply patches/composer.patch
composer dump-autoload --no-interaction
```

That's quite cumbersome.

And `git apply patches/composer.patch` doesn't even work for me, because `wordpress-tests-lib` is not in `/tmp` on macOS.

Reply:
I was just going to change to run PHP 7.3 locally, as the issue is only in 7.4...

This does open the question, if we run tests in PHP 7.2.

Reply:
I don't think we need to start running tests on PHP 7.2.

Also not fond of having to downgrade PHP locally now to work around this limitation :/

Reply:
My worry here, is if we upgrade on PHPUnit version locally and run our tools as PHP 7.4, that at some point, the composer.lock file will be checked in. 

It is worth noting, we can't currently run the PHPUnit tests in PHP locally either. Changing to PHP 7.3 isnt that bad.

Reply:
> It is worth noting, we can't currently run the PHPUnit tests in PHP locally either.

Can you elaborate? I currently run them locally all the time with no issues, using PHP 7.4.

Reply:
>Can you elaborate? I currently run them locally all the time with no issues, using PHP 7.4.

You can't run them in PHP 8.0. 

Changing PHP version is very easy with homebrew. https://dcblog.dev/switching-between-php-versions-with-homebrew. That or you patch it locally or ignore the warnings.

Reply:
Then we'd need to communicate this to all contributors that they can only run tests locally on PHP 7.3. Which is a step backward from today.

> That or you patch it locally

Gonna be cumbersome having to manually patch it every time I work on the codebase and ensuring I don't accidentally commit the changes when switching branches and submitting PRs.

> or ignore the warnings.

They're unfortunately errors.

-----

IMO we should also look into the other suggestion kindly provided at https://github.com/Yoast/PHPUnit-Polyfills/issues/45#issuecomment-877416085

A third option might be to patch PHPUnit 6 instead so that these errors no longer happen.

If we can't find a suitable solution that does not affect DX I'm inclined to just remove the return type annotations for the time being and save those for a future change until we can figure out a better solution.

Reply:
I did read the comment 😄 

Which is why I followed her advice 

> Or keep the lock at PHPUnit 6.x and upgrade the PHPUnit version in your CI script for any test runs on PHP >= 7.4.

This is exactly what I did and what this PR does. 

Are you recommend that do option 1. 

> Either "lock" PHPUnit at PHPUnit 7.x and downgrade the PHPUnit version in your CI script for the test run against PHP 7.0.

If so, I think that has some bigger issues. If we commit PHPUnit 7.x and it's dependencies to lock file, it would mean, everywhere we need to code to run in PHP 7.0, like building the plugin, we would need to downgrade PHPUnit again. This is because overlapping dependency, might make errors. 

Another idea would be maintain two composer files. `composer-dev.json` and `composer.json`

composer-dev.json would not lock the system required down to PHP 7.0. Meaning, composer install would pick the correct version of PHPUnit. 

composer.json would only contain what is need to build the plugin for E2E / release. 

Or we could just remove the return types....

Reply:
Let's remove the return types for now and deal with that in a new ticket.

Reply:
@swissspidy So I did some more research and seems like the issue is really param hints are the issue. Should I revert those and keep the return type hinting?